src/thumbnailer.h

```diff
@@ -64,17 +65,29 @@ class Thumbnailer {
 
   // Returns the PSNR value of a re-encoded WebPPicture at given config.
   // Returns -1 on errors.
-  int GetPSNR(WebPPicture* const pic, const WebPConfig& config);
+  float GetPSNR(WebPPicture* const pic, const WebPConfig& config);
+
+  // Returns the size of a re-encoded WebPPicture at given config in bytes.
+  // Returns -1 on errors.
+  int GetSize(WebPPicture* const pic, const WebPConfig& config);
 
   // Generates the animation so that all frames have similar PSNR (all) values.
   // In case of failure, takes the animation generated by GenerateAnimation() as
   // result.
   Status GenerateAnimationEqualPSNR(WebPData* const webp_data);
 
+  // Tries near-lossless to encode each frame. In case of failure, keeps the
+  // latest lossy encoded frame.
+  Status TryNearLossless(WebPData* const webp_data);
+
   Status SetLoopCount(WebPData* const webp_data);
 
-  // List of final quality for each frames.
-  std::vector<int> final_qualities;
+  struct FrameResult {
+    int size;
+    int quality;
+  };
+  // Vector of frames's final results.
+  std::vector<FrameResult> results_;
```

Comment:
does this have to be public?

Reply:
Also is there a reason why this is a separate `vector` and not part of `FrameData`?

Reply:
the `results_ vector` is probably unnecessary, I will move the `size` and `quality` to `FrameData`.